wandb/sdk/launch/registry/google_artifact_registry.py

```diff
@@ -169,3 +169,32 @@ def get_repo_uri(self) -> str:
             f"{self.environment.region}-docker.pkg.dev/"
             f"{self.environment.project}/{self.repository}/{self.image_name}"
         )
+
+    def check_image_exists(self, image_uri: str) -> bool:
+        """Check if the image exists.
+
+        Arguments:
+            image_uri: The image URI.
+
+        Returns:
+            True if the image exists, False otherwise.
+        """
+        _logger.info(
+            f"Checking if image {image_uri} exists. In Google Artifact Registry {self.uri}."
+        )
+        repo_uri, _ = image_uri.split(":")
+        if repo_uri != self.get_repo_uri():
+            raise LaunchError(
+                f"The image {image_uri} does not belong to the Google Artifact "
+                f"Repository {self.get_repo_uri()}."
+            )
+        credentials = self.environment.get_credentials()
+        request = google.cloud.artifactregistry.GetTagRequest(parent=image_uri)
+        client = google.cloud.artifactregistry.ArtifactRegistryClient(
+            credentials=credentials
+        )
+        try:
+            client.get_tag(request=request)
+            return True
+        except google.api_core.exceptions.NotFound:
+            return False
```

Comment:
Did you test this out for real?

Reply:
No, I'm not set up, I return False for now and put a TODO for this, but I'm not set up on GCP and don't think this can be done prior to release